### PR TITLE
Delay loading spinner

### LINKED ISSUE #5345: Delay the loading spinner
As discussed, frequently the loading spinner flashes quickly before content is rendered. To avoid this flash, we should delay the loading spinner by ~300ms before showing it. We should delay by default, but allow specific instances to configure their own delay.

### PR DESCRIPTION
**Pull request by lovincyrus**
Reviewers - Check out this branch locally, and please let me know if any of the hot path's spinners shouldn't be delayed.

The goal is to eliminate the flashing/flickering spinners on page load. To do that, we only show the spinner if the loading state takes longer than the specified delay.

I've added two new helpers, `DelayedSpinner.svelte` ~~and `spinner-utils.ts`~~, to efficiently manage different spinner states. ~~Use `spinner-utils.ts` for managing loading states in useQuery.~~ `DelayedSpinner.svelte` works best for various other conditions, especially within inline tertiary conditionals.

The delayed spinner is handy for preventing the loading state from flickering. We've added a default of 300ms delay before actually displaying the spinner.

closes #5345

ref https://github.com/rilldata/rill/issues/1567

- [x] revise existing `<Spinner/>`

Before

https://github.com/user-attachments/assets/6ddeb861-1256-42b3-b7ab-aa66ae590884

After

https://github.com/user-attachments/assets/83156c49-eb42-4a14-9b46-14cb445df85a

Also notably snappy big number (no flickering spinner and no data text)

https://github.com/user-attachments/assets/c639a5c4-7b57-47bd-90aa-7238b73e4e5a

### CODE REVIEW

File: web-admin/src/features/alerts/listing/AlertsTable.svelte [hunk 1]
```diff
@@ -11,12 +11,17 @@
...
+  import { isLoadingWithTimeout } from "@rilldata/web-common/features/entity-management/spinner-utils";
 
   export let organization: string;
   export let project: string;
 
   $: alerts = useAlerts($runtime.instanceId);
 
+  const loadingWithTimeout = isLoadingWithTimeout();
+  $: loadingWithTimeout.setLoading($alerts.isLoading);
+  $: loadingAlerts = $loadingWithTimeout;
```
Comment (ericpgreen2) [inline web-admin/src/features/alerts/listing/AlertsTable.svelte:23]: Thoughts on updating the `isLoadingWithTimeout()` function to make this a one-liner?

Comment (lovincyrus) [reply web-admin/src/features/alerts/listing/AlertsTable.svelte:23]: sure, I can do that. This helps us remove the boilerplate-y code.

`$: loading = isLoadingWithTimeout(isLoading)` is pretty straight forward.

**Comment by lovincyrus** [comment]
In response to https://github.com/rilldata/rill/pull/5351#discussion_r1700997760, I tried simplifying the code into a one-liner without success. However, I managed to boil it down to `<DelayedSpinner />` so we only have one abstraction going forward.

We could integrate the delayed spinner within nested conditional statements more efficiently:

```svelte
{#if isLoading}
  <DelayedSpinner isLoading={isLoading} />
{:else if $isError}
  <Error />
{:else if $isSuccess}
  {#if $data.length === 0}
    <Empty />
  {:else}
    <Data />
  {/if}
{/if}
```

Curious for your thoughts @ericpgreen2

File: web-admin/src/features/alerts/listing/AlertsTable.svelte (hunk 1 above)
Comment (lovincyrus) [reply web-admin/src/features/alerts/listing/AlertsTable.svelte]: See: https://github.com/rilldata/rill/pull/5351#issuecomment-2266283116

**Comment by jkhwu** [comment]
This looks like an improvement to me! Thank you.

File: web-common/src/features/entity-management/DelayedSpinner.svelte [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    }
+  });
+
+  onDestroy(() => {
+    clearTimeout(timeoutId);
+  });
+</script>
+
+{#if $showSpinner}
+  <Spinner status={EntityStatus.Running} {duration} {size} />
```
Comment (AdityaHegde) [inline web-common/src/features/entity-management/DelayedSpinner.svelte:41]: We should get rid of `EntityStatus`, it was a concept when we had nodejs backend. Perhaps copy over the contents of `Spinner.svelte` in here as a start?

File: web-common/src/layout/BlockingOverlayContainer.svelte [hunk 3]
```diff
@@ -32,7 +32,10 @@
         style:font-size="48px"
       >
         <div class="on" style="--length: {2000 + Math.random() * 5000}ms;">
-          <Spinner {status} duration={300 + Math.random() * 200} />
+          <DelayedSpinner
```
Comment (AdityaHegde) [inline web-common/src/layout/BlockingOverlayContainer.svelte:35]: Does it make sense here? If a blocking overlay is present it should always have a spinner no?

File: web-common/src/features/entity-management/DelayedSpinner.svelte [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    } else {
+      showSpinner.set(false);
+    }
+  }
+
+  onMount(() => {
+    if (isLoading) {
+      timeoutId = setTimeout(() => showSpinner.set(true), delay);
+    }
+  });
```
Comment (ericpgreen2) [inline web-common/src/features/entity-management/DelayedSpinner.svelte:33]: Looks like this is redundant to the reactive statement, so it can be removed

**Review by ericpgreen2 — APPROVED** [review]
Sweet 👍. I agree with Aditya that we should deprecate the `EntityStatus` concept, but that's slightly wider in scope, so perhaps that merits a standalone follow-up PR. Approving so you can merge at your discretion.

File: web-common/src/features/entity-management/DelayedSpinner.svelte (hunk 2 above)
Comment (lovincyrus) [reply web-common/src/features/entity-management/DelayedSpinner.svelte:35]: Will deprecate the `EntityStatus` in a follow-up.

File: web-common/src/layout/BlockingOverlayContainer.svelte (hunk 3 above)
Comment (lovincyrus) [reply web-common/src/layout/BlockingOverlayContainer.svelte]: https://github.com/rilldata/rill/pull/5351/commits/e352da7bde1f34e6c06100ed614ac52892aa069c

**Pull request merged**